tests/st/policy/test_felix_gateway.py

```diff
@@ -0,0 +1,510 @@
+        ('deny', 'allow')
+    ])
+    def test_conflicting_ingress_and_egress_policy(self, in_action, out_action):
+        # If there is policy on the ingress and egress interface then both should
+        # get applied and 'deny' should win.
+        self.add_host_iface()
+        self.add_gateway_external_iface()
+        self.add_gateway_internal_iface()
+
+        self.add_policy({
+            'apiVersion': 'v1',
+            'kind': 'policy',
+            'metadata': {'name': 'port80-int'},
+            'spec': {
+                'order': 10,
+                'ingress': [
+                    {
+                        'action': in_action
+                    },
+                ],
+                'egress': [],
+                'selector': 'role == "gateway-int"'
+            }
+        })
+        self.add_policy({
+            'apiVersion': 'v1',
+            'kind': 'policy',
+            'metadata': {'name': 'port80-ext'},
+            'spec': {
+                'order': 10,
+                'ingress': [],
+                'egress': [
+                    {
+                        'action': out_action
+                    },
+                ],
+                'selector': 'role == "gateway-ext"'
+            }
+        })
+        retry_until_success(self.assert_host_can_not_curl_ext, 3)
+
+    def add_policy(self, policy_data):
+        self._apply_resources(policy_data, self.gateway)
+
+    def add_gateway_internal_iface(self):
+        host_endpoint_data = {
+            'apiVersion': 'v1',
+            'kind': 'hostEndpoint',
+            'metadata': {
+                'name': 'gw-int',
+                'node': self.gateway_hostname,
+                'labels': {'role': 'gateway-int'}
+            },
+            'spec': {
+                'interfaceName': 'eth0'
+            }
+        }
+        self._apply_resources(host_endpoint_data, self.gateway)
+
+    def add_gateway_external_iface(self):
+        host_endpoint_data = {
+            'apiVersion': 'v1',
+            'kind': 'hostEndpoint',
+            'metadata': {
+                'name': 'gw-ext',
+                'node': self.gateway_hostname,
+                'labels': {'role': 'gateway-ext'}
+            },
+            'spec': {
+                'interfaceName': 'eth1'
+            }
+        }
+        self._apply_resources(host_endpoint_data, self.gateway)
+
+    def add_host_iface(self):
+        host_endpoint_data = {
+            'apiVersion': 'v1',
+            'kind': 'hostEndpoint',
+            'metadata': {
+                'name': 'host-int',
+                'node': self.host_hostname,
+                'labels': {'role': 'host'}
+            },
+            'spec': {
+                'interfaceName': 'eth0',
+                'expectedIPs': [self.get_container_ip('cali-st-host')],
+            }
+        }
+        self._apply_resources(host_endpoint_data, self.gateway)
+
+    @staticmethod
+    def _apply_resources(resources, host):
```

Comment:
We should move this into the FV libraries.  Not necessarily something you should do for this PR, but someone, soon.